### ISSUE TITLE
GameObjects with a trigger collider can phase through the world.

### ISSUE THREAD

**Issue by Fortune117** [Facepunch/sbox-issues#5027]
### Describe the bug

In my game I have two objects, one with a box collider component that has `IsTrigger` set to true:
![image](https://github.com/Facepunch/sbox-issues/assets/7847372/91612112-60d9-4e4d-b86e-1315b5f5a0ce)

The other is otherwise identical but does not have the box collider:
![image](https://github.com/Facepunch/sbox-issues/assets/7847372/8778c146-8eee-4fa0-aa19-262bc15ad05d)

When I start the game the game object with the box collider will fall through the world, while the one without a collider doesn't. 

However, if I start the game with the game object raised above the floor it doesn't fall through the world:
![image](https://github.com/Facepunch/sbox-issues/assets/7847372/33fa606e-8a8a-4c21-a0f3-087d5b88a971)

### To Reproduce

1. Add this game object to a scene, the model doesn't really matter

```
[{"Id":"958ece3e-b92b-4a23-a319-4e0ad6d05f07","Name":"mushroom_01 (3)","Position":"-370.9852,492,0","Enabled":true,"Components":[{"__type":"Prop","Health":0,"IsStatic":false,"Model":"models/props/food/mushroom/mushroom_01.vmdl","OnPropBreak":null,"OnPropTakeDamage":null,"ShowCreatedComponents":false},{"__type":"BoxCollider","__enabled":false,"Center":"0,0,0","IsTrigger":true,"Scale":"10,10,10","Static":false,"Surface":null}]}]
```

2. Enable the trigger box collider and place it on the floor.
3. Should fall through the world when the game starts.

### Expected behavior

The trigger collider shouldn't affect its behavior in this way.

**Comment by aylaylay** [comment]
I copied the game object and it's not falling through the floor?

**Comment by aylaylay** [comment]
Maybe this was fixed recently?

**Comment by Fortune117** [comment]
I did some testing, seems to only be with map physics. I've uploaded a repo with a map component and a flat plane collider. There are two coins in the scene, one closer to the ground than the other. One will consistently fall through the floor when using the map collider, but neither falls through when there's just a plane.

I also found that the game object having a Box Collider component marked as trigger does in fact cause it to fall through the floor, so it's not just because of the small colliders. If you disable the box colliders on the coins, neither of them fall through the map instance.

[bug_repo.zip](https://github.com/Facepunch/sbox-issues/files/14783050/bug_repo.zip)
